enhancements/sig-architecture/20-taint-toleration/README.md

```diff
@@ -0,0 +1,312 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management/website/)
+
+## Summary
+
+The proposed work would define Taints and Tolerations in placement APIs to allow users to control the selection of managed clusters more flexibly, and evict selected unhealthy/not-reporting clusters after a certain period.
+
+## Motivation
+
+For the new placement enhancement need of supporting filtering unhealthy/not-reporting clusters and keep workloads from being placed in unhealthy or unreachable clusters, we may consider the similar logic of taint/toleration in Kubernetes. Basically, there are two things we could do:
+
+1) Implement taint/toleration in placement apis.
+2) Add a new controller to map states of clusters to taints automatically.
+
+Taints and tolerations would work together to allow users to control the selection of managed clusters more flexibly, and evict selected clusters after a certain period. Specifically, Taints are properties of managed clusters, they allow a placement to repel a set of managed clusters. Tolerations are applied to placements, and allow the managed clusters with matching taints to be scheduled onto placements.
+
+### User Stories
+
+Story 1: Users can prevent some workloads from being scheduled to those managed clusters.
+
+Story 2: Users schedule some workloads to certain managed clusters.
+
+Story 3: When a managed cluster gets offline, the system can make applications deployed on this cluster to be transferred to another available managed cluster immediately. 
+
+Story 4: Users are able to evict workloads deployed on a cluster without modifying or being aware of existing placements’ details.
+
+Story 5: When a managed cluster gets offline, users can make applications deployed on this cluster to be transferred to another available managed cluster after a tolerated time.
+
+## Proposal
+
+### API repo
+
+#### ManagedClusterSpec
+In `ManagedClusterSpec`, we can add `taints` which has the similar structure with taints in Kubernetes. But to keep things simple at the beginning, we can remove `effect` which is in Kubernetes' taint while only keep `key` and `value` .
+
+```go
+type ManagedClusterSpec struct {
+   ManagedClusterClientConfigs []ClientConfig `json:"managedClusterClientConfigs,omitempty"`
+   HubAcceptsClient bool `json:"hubAcceptsClient"`
+   LeaseDurationSeconds int32 `json:"leaseDurationSeconds,omitempty"`
+   // Taints is a property of managed cluster that allow the cluster to be repelled when scheduling.
+   // +optional
+   Taints []Taint `json:"taints,omitempty"`
+}
+ 
+type Taint struct {
+   // Required. The taint key to be applied to a cluster.
+   Key string `json:"key"`
+   // The taint value corresponding to the taint key.
+   // +optional
+   Value string `json:"value,omitempty"`
+   // TimeAdded represents the time at which the taint was added.
```

Comment:
indicate what is nil mean here.

Reply:
> indicate what is nil mean here.

looks like this comment still need addressing. Don't we always know the time the taint was added and if so, it doesn't need to be a pointer, right?